src/core.c/Str.pm6

```diff
@@ -248,6 +248,23 @@ my class Str does Stringy { # declared in BOOTSTRAP
         )
     }
 
+    multi method substr-between(Str:D $before, Str:D $after, Int:D $pos = 0) {
+        my int $left = nqp::index(self, $before, $pos);
+        if $left > -1 {
+            my int $right = nqp::index(self, $after, $left);
+            if $right > -1 {
+                my int $offset = $left + nqp::chars($before);
+                nqp::substr(self, $offset, $right - $offset)
```

Comment:
This will not work on overlapping strings:

src: "0123456789";
before: "2345"
after: "456"

This will result in `$left=2`, `$right=4`, `$offset=6` -> `$right - $offset = -2`. Perhaps `$right` should be obtained on `$left + nqp::chars($before)`?

Reply:
Perhaps  :-)